### PR TITLE
Exposing isInfinity function

### PR DESCRIPTION
**Pull request by fedejinich**
On `secp256k1` library, there is an [edge case](https://github.com/bitcoin-core/secp256k1/issues/109) when recovering a Pub Key (that its point is infinity).

I'm exposing a native function to check if a recovery will lead to this particular case, so we can handle it properly on rskj side.

PD: I've also improved some `README.md` and `build.gradle` descriptions.

### CODE REVIEW

**Review by patogallaiovlabs — DISMISSED** [review]
LGTM

**Review by patogallaiovlabs — COMMENTED** [review]
Why did you remove the assertEquals functions?

**Comment by fedejinich** [comment]
Well the fact is they don't add any significant value, they were just reimplementing JUinit assertion library and also they enforce you to test things in their way (with no reason). 

Plus I prefer throwing a custom exception (such as NativeSecp256k1Exception instead of AssertFailure).

**Review by patogallaiovlabs — DISMISSED** [review]
LGTM

File: src/main/java/org/bitcoin/NativeSecp256k1Util.java [hunk 1]
```diff
@@ -16,38 +16,78 @@
...
+        return String.join("", Collections.nCopies(64, "0")).toLowerCase();
     }
 
-    public static void checkArgument(boolean expression) {
-        if (!expression) {
-            throw new IllegalArgumentException();
-        }
+    // Testing
+    public static String random32String() {
+        return "CF80CD8AED482D5D1527D7DC72FCEFF84E6326592848447D2DC0B0E87DFC9A90".toLowerCase();
```
Comment (juli) [inline src/main/java/org/bitcoin/NativeSecp256k1Util.java]: ?

File: src/main/java/org/bitcoin/NativeSecp256k1Util.java [hunk 2]
```diff
@@ -16,38 +16,78 @@
...
+     * @param s
+     * @param fixed
+     * @return
+     */
+    public static byte[] concatenate(byte[] r, byte[] s, boolean fixed) {
+        byte[] allByteArray = new byte[fixed ? 64 : r.length + s.length];
+        ByteBuffer buff = ByteBuffer.wrap(allByteArray);
+        if(fixed) {
+            for (int i = r.length; i < 32; i++) {
+                buff.put((byte) 0);
```
Comment (juli) [inline src/main/java/org/bitcoin/NativeSecp256k1Util.java]: are you sure you need to do this?

File: src/main/java/org/bitcoin/NativeSecp256k1Util.java [hunk 3]
```diff
@@ -16,38 +16,78 @@
...
+        return sBytes.length > 32 && fixed ? sBytes.length - 32 : 0;
     }
 
-    public static class AssertFailException extends Exception {
-      public AssertFailException(String message) {
-        super( message );
-      }
+
+    // Testing
+    public static String zero32String() {
```
Comment (juli) [inline src/main/java/org/bitcoin/NativeSecp256k1Util.java]: remove

**Review by juli — COMMENTED** [review]
remove testing code

File: src/main/java/org/bitcoin/NativeSecp256k1Util.java (hunk 2 above)
Comment (fedejinich) [reply src/main/java/org/bitcoin/NativeSecp256k1Util.java]: unnecesary, removed 👍

File: src/main/java/org/bitcoin/NativeSecp256k1Util.java (hunk 1 above)
Comment (fedejinich) [reply src/main/java/org/bitcoin/NativeSecp256k1Util.java]: removed

File: src/main/java/org/bitcoin/NativeSecp256k1Util.java (hunk 3 above)
Comment (fedejinich) [reply src/main/java/org/bitcoin/NativeSecp256k1Util.java]: removed

**Comment by donequis** [comment]
This seems to force the client to perform two recoveries instead of one. It should return the same key as the java implementation when q is infinity

**Review by donequis — APPROVED** [review]
We discussed it privately and decided that it's not worth the performance improvements changing this approach. It should do ok, and will be reviewed again once the RSKj PR is ready

**Review by aeidelman — APPROVED** [review]

**Pull request merged**